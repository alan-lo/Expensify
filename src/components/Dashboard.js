import React from 'react';

const ExpenseDashboardPage = ()=>{
  return(
    <div>Dashboard</div>
  );
}

export default ExpenseDashboardPage;