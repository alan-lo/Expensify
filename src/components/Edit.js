import React from 'react';

const Edit = (props)=>{
  console.log(props.match.params.id);
  return(
    <div>Edit Page</div>
  );
}

export default Edit;
