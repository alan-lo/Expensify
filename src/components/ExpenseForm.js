import React , {Component} from 'react';
import moment from 'moment';
import 'react-dates/initialize'
import {SingleDatePicker} from 'react-dates';
import 'react-dates/lib/css/_datepicker.css';

class ExpenseForm extends Component{
  state = {
    description: '',
    note: '',
    amount: '',
    createdAt: moment(),
    focused: false
  }

  onDescriptionChange = (e)=>{
    const description = e.target.value;
    this.setState((prevState)=>({description}))
  }

  onNoteChange = (e)=>{
    const note = e.target.value; 
    this.setState((prevState)=>({note}));
  } 

  onAmountChange= (e)=>{
    const amount = e.target.value;
    if (!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)) {
      this.setState((prevState)=>({amount}));
    }
  }

  onDateChange=(createdAt)=>{
    if (createdAt){
      this.setState(()=>({createdAt}));
    }
  }

  onFocusChange=({focused})=>{
    this.setState(()=>({focused}));   
  }

  render(){
    return(
      <div>
        <form>
        <input 
          type="text" 
          placeholder="description" 
          autoFocus
          value={this.state.description}
          onChange={this.onDescriptionChange}
        />
        <input 
          type="text" 
          placeholder="amount"
          value={this.state.amount}
          onChange={this.onAmountChange}
        />
        <SingleDatePicker
          date={this.state.createdAt}
          onDateChange={this.onDateChange} 
          focused={this.state.focused} 
          onFocusChange={this.onFocusChange}
          numberOfMonths={1} 
          isOutsideRange={()=> false}
        />
        <textarea 
          placeholder="Note"
          value= {this.state.note}
          onChange={this.onNoteChange}
        >
        </textarea>

        <button>Add Expense</button>
        </form>
      </div>
    );
  }
}

export default ExpenseForm;