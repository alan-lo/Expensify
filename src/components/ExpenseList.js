import React from 'react';
import {connect} from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expense_selector';

const ExpenseList = (props)=>{
  return (
    <div>
      <div>Expense List</div>
      {props.expenses.map((expense)=>{
        return (
          <ExpenseListItem key={expense.id} {...expense}/>
        );
      })}
    </div>
  );
}

const mapStateToProps = (state)=>({
  expenses: selectExpenses(state.expenses, state.filters)
})


export default connect(mapStateToProps)(ExpenseList);