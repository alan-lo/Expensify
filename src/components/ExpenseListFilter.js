import React, {Component} from 'react';
import {connect} from 'react-redux';
import  {setTextFilter, sortByAmount , sortByDate, setEndDate, setStartDate} from '../actions/filters_action';
import 'react-dates/initialize';
import {DateRangePicker} from 'react-dates';

export class ExpenseListFilter extends Component {
  state = {
    focused: null
  }

  onDateChange = ({startDate, endDate})=>{
    this.props.setStartDate(startDate);
    this.props.setEndDate(endDate);
  }

  onFocusChange = (focused)=>{
    this.setState(()=>({focused}))
  }

  onTextChange = (e)=>{
    this.props.setTextFilter(e.target.value);
  }

  onSortChange = (e)=>{
    if (e.target.value == 'date'){
      this.props.sortByDate();
    }else if (e.target.value =='amount'){
      this.props.sortByAmount();
    }
  }

  render(){
    return (
      <div>
        <input 
          type="text" 
          value={this.props.filters.text}
          onChange={this.onTextChange} 
        />
        <select
          value={this.props.filters.sortBy}
          onChange={this.onSortChange}
        >
          <option value="date">Date</option>
          <option value="amount">Amount</option>
        </select>
        <DateRangePicker
          startDate={this.props.filters.startDate} 
          startDateId='start'
          endDate={this.props.filters.endDate} 
          endDateId='end'        
          onDatesChange={this.onDateChange} 
          focusedInput={this.state.focused}
          onFocusChange={this.onFocusChange} 
          numberOfMonths={1}
          isOutsideRange={()=>false}
          showClearDates={true}
        />
      </div>
    );
  }

}

const mapStateToProps = (state)=>({
    filters: state.filters
})

const mapDispatchToProps = (dispatch)=>({
  setTextFilter : (text)=> dispatch(setTextFilter(text)),
  sortByDate    : ()=> dispatch(sortByDate()),
  sortByAmount  : ()=> dispatch(sortByAmount()),
  setStartDate  : (startDate) => dispatch(setStartDate(startDate)),
  setEndDate    : (endDate) => dispatch(setStartDate(endDate))
});
export default connect(mapStateToProps)(ExpenseListFilter);