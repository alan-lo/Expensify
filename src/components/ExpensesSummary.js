import React, {Component} from 'react';
import {connect} from 'react-redux';
import selectExpensesTotal from '../selectors/expense_total';
import selectExpenses from '../selectors/expense_selector';
import numeral from 'numeral';

export const ExpensesSummary = ({expenseCount, expensesTotal})=>{
    const expenseWord = expenseCount > 1 ? 'expenses' : 'expense';
    const formattedExpensesTotal = numeral(expensesTotal/ 100).format('$0,0.00')
    return (
      <div>
        <h3>Viewing {expenseCount} {expenseWord} totalling {formattedExpensesTotal}</h3>
      </div>
    );
}

const mapStateToProps = (state) => {
  const getVisbleExpense = selectExpenses(state.expenses, state.filters);
  return {
     expensesTotal: selectExpensesTotal(getVisbleExpense),
     expenseCount: getVisbleExpense.length
  };
};

export default connect(mapStateToProps)(ExpensesSummary);