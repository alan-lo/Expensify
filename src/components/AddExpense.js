import React from 'react';
import ExpenseForm from './ExpenseForm';
import {connect} from 'react-redux';
import {addExpense} from '../actions/expenses_action';

const AddExpense = (props)=>{
  return(
    <div>
      <h2>Add Expense Page</h2>
      <ExpenseForm onSubmit={
        (expense)=>{
          props.dispatch(addExpense(expense))
          props.history.push('/')
        }
      }
      />
    </div>
  );
}

export default connect()(AddExpense);
