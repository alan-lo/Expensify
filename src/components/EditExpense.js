import React from 'react';
import ExpenseForm from './ExpenseForm';
import {editExpense, removeExpense} from '../actions/expenses_action';
import {connect} from 'react-redux';

export class Edit extends React.Component{

  onSubmitEdit = (expense) => {
    console.log('updated', expense);
    this.props.editExpense(this.props.expense.id , expense);
    this.props.history.push('/');
  }

  onClickRemove = () => {
    this.props.removeExpense(this.props.expense);
    this.props.history.push('/');
  }

  render(){
    return(
      <div>
        <ExpenseForm 
          onSubmit={this.onSubmitEdit}
          expense = {this.props.expense} 
        />
        <button onClick={this.onClickRemove}>Remove</button>
      </div>
    );
  }
}

const mapDispatchToProps = (dispatch)=> ({
  removeExpense : (expense) => {
    dispatch(removeExpense({id: expense.id}))
  }
  ,
  editExpense : (id, expense) => {  
    dispatch(editExpense(id ,expense))
  }
})

const mapStateToProps = (state, props)=>{
  return {
    expense: state.expenses.find((expense)=>{
      return expense.id === props.match.params.id
    })
  };
}
export default connect(mapStateToProps, mapDispatchToProps)(Edit);
