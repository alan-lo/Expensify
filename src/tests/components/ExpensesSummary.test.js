import React from 'react';
import {shallow} from 'enzyme';
import {ExpensesSummary} from '../../components/ExpensesSummary';

test('render correctly expenses summary with one expense', ()=> {
 const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={123.22}/>);
 expect(wrapper).toMatchSnapshot();
})


test('render correctly expenses summary with multiple expenses', ()=>{
  const wrapper = shallow(<ExpensesSummary expenseCount={2} expensesTotal={223.22}/>);
  expect(wrapper).toMatchSnapshot();
})