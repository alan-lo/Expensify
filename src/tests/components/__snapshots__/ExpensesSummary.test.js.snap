// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly expenses summary with multiple expenses 1`] = `
<div>
  <h3>
    Viewing 
    2
     
    expenses
     totalling 
    $2.23
  </h3>
</div>
`;

exports[`render correctly expenses summary with one expense 1`] = `
<div>
  <h3>
    Viewing 
    1
     
    expense
     totalling 
    $1.23
  </h3>
</div>
`;
